pkg/controller/core.oam.dev/v1alpha2/applicationconfiguration/revision_enable_test.go

```diff
@@ -341,3 +343,165 @@ var _ = Describe("Test ApplicationConfiguration Component Revision Enabled trait
 	})
 
 })
+
+var _ = Describe("Test Component Revision Enabled with custom component revision hook", func() {
+	const (
+		namespace = "revision-enable-test2"
+		compName  = "revision-test-comp2"
+	)
+	var (
+		ctx       = context.Background()
+		component v1alpha2.Component
+		ns        = corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+	)
+
+	BeforeEach(func() {})
+
+	AfterEach(func() {
+		// delete the namespace with all its resources
+		Expect(k8sClient.Delete(ctx, &ns, client.PropagationPolicy(metav1.DeletePropagationForeground))).
+			Should(SatisfyAny(BeNil(), &util.NotFoundMatcher{}))
+	})
+
+	FIt("custom component change revision lead to revision difference, it should not loop infinitely create", func() {
```

Comment:
have you tried this test without the fix?

Reply:
of course